runtime/gc_vlhgc/ConfigurationIncrementalGenerational.cpp

```diff
@@ -142,18 +142,25 @@ MM_ConfigurationIncrementalGenerational::createHeapWithManager(MM_EnvironmentBas
 	 */
 	vm->indexableObjectLayout = J9IndexableObjectLayout_DataAddr_Arraylet;
 #if defined(J9VM_GC_SPARSE_HEAP_ALLOCATION)
-	if (extensions->isVirtualLargeObjectHeapRequested) {
+	/* set off-heap enabled as default for balanced GC */
+	extensions->isVirtualLargeObjectHeapEnabled = true;
+
+	if (extensions->virtualLargeObjectHeap._wasSpecified) {
+		extensions->isVirtualLargeObjectHeapEnabled = extensions->virtualLargeObjectHeap._valueSpecified;
+	}
+
+	if (extensions->isVirtualLargeObjectHeapEnabled) {
 		/* Create off-heap */
 		MM_SparseVirtualMemory *largeObjectVirtualMemory = MM_SparseVirtualMemory::newInstance(env, OMRMEM_CATEGORY_MM_RUNTIME_HEAP, heap);
 		if (NULL != largeObjectVirtualMemory) {
 			extensions->largeObjectVirtualMemory = largeObjectVirtualMemory;
 			extensions->indexableObjectModel.setEnableVirtualLargeObjectHeap(true);
-			extensions->isVirtualLargeObjectHeapEnabled = true;
 			/* Overriding the original assumption that Balanced has arraylets. */
 			vm->indexableObjectLayout = J9IndexableObjectLayout_DataAddr_NoArraylet;
 			/* reset vm->unsafeIndexableHeaderSize for off-heap case */
 			vm->unsafeIndexableHeaderSize = 0;
 		} else {
+			extensions->isVirtualLargeObjectHeapEnabled = false;
```

Comment:
minor, but do we want to reset it? it'a configuration flag, not some structure we want to clean up.

 @dmitripivkine, opinion?

Reply:
If offheap is requested (by default or explicitly) but Sparse Virtual Memory can not be initialized we should fail to start JVM. In this case reseting `isVirtualLargeObjectHeapEnabled` is irrelevant. I think we should not do it here.